src/Core/Domain/Carrier/ValueObject/CarrierRangePrice.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Carrier\ValueObject;
+
+use PrestaShop\Decimal\DecimalNumber;
+
+/**
+ * Price by Range and Zone for Carriers
+ */
+class CarrierRangePrice
+{
+    private DecimalNumber $from;
+    private DecimalNumber $to;
+    private DecimalNumber $price;
+
+    public function __construct(
+        float|string $from,
+        float|string $to,
```

Comment:
When we rely on DecimalNumber the input should always be a string Mabe one day It would be a good improvement to allow `string|DecimalNumber` But the idea is to ban the usage of float, so it shouldn't be accepted as a valid input

Note: you can still use DecimalNumber in the query result as we consider it a native type